### PR TITLE
Implemented #247, #250, #259, and upgrade all nuget packages

### PR DESCRIPTION
**Pull request by aiviet**
Implemented #247, #250, #259, and upgrade all nuget packages

### CODE REVIEW

File: MediaManager.Abstractions/Enums/MediaTrackType.cs [hunk 1]
```diff
@@ -0,0 +1,28 @@
+namespace Plugin.MediaManager.Abstractions.Enums
+{
+    public enum MediaTrackType
```
Comment (martijn00) [inline MediaManager.Abstractions/Enums/MediaTrackType.cs]: What is this type supposed to do? There is already an type to say what video or audio it is.

File: MediaManager.Abstractions/IMediaTrackInfo.cs [hunk 2]
```diff
@@ -0,0 +1,20 @@
+
+using System.ComponentModel;
+using Plugin.MediaManager.Abstractions.Enums;
+
+namespace Plugin.MediaManager.Abstractions
+{
+    public interface IMediaTrackInfo : INotifyPropertyChanged
```
Comment (martijn00) [inline MediaManager.Abstractions/IMediaTrackInfo.cs]: What are the extra properties for? There already is a TrackInfo class.

File: MediaManager.Abstractions/IVideoPlayer.cs [hunk 3]
```diff
@@ -21,5 +23,18 @@ public interface IVideoPlayer : IPlaybackManager
         /// The aspect mode of the video
         /// </summary>
         VideoAspectMode AspectMode { get; set; }
+        
+        int TrackCount { get; }
```
Comment (martijn00) [inline MediaManager.Abstractions/IVideoPlayer.cs]: Can't you get this through the Queue?

**Comment by martijn00** [comment]
Thanks for your contribution! It is really hard to see why some of the changes where made. It would be really helpful to split this up in smaller changes and different PR's so it's much easier to merge and review.

In general me and @modplug are reviewing the current API and will work on a new vNext based on Net Standard. It would be great to already take your changes in account.

File: MediaManager.Abstractions/Enums/MediaTrackType.cs (hunk 1 above)
Comment (aiviet) [reply MediaManager.Abstractions/Enums/MediaTrackType.cs]: Are you referring to the existing MediaPlaybackType enum of Video and Audio?  Anyway the new MediaTrackType is to abstract these Android and iOS enum for the purpose of selecting audio channel of a video:
https://developer.xamarin.com/api/type/Android.Media.MediaTrackType/
https://developer.xamarin.com/api/type/MonoTouch.AVFoundation.AVMediaType/

File: MediaManager.Abstractions/IMediaTrackInfo.cs (hunk 2 above)
Comment (aiviet) [reply MediaManager.Abstractions/IMediaTrackInfo.cs]: Are you referring to the Android TrackInfo? https://developer.xamarin.com/api/type/Android.Media.MediaPlayer+TrackInfo/
This new class is to abstract TrackInfo at PCL level. Need this to implement the selection of audio channel.

File: MediaManager.Abstractions/IVideoPlayer.cs (hunk 3 above)
Comment (aiviet) [reply MediaManager.Abstractions/IVideoPlayer.cs]: TrackCount is to store the number of available tracks in a video:
Android: https://developer.xamarin.com/api/member/Android.Media.MediaPlayer.GetTrackInfo/
iOS: https://developer.xamarin.com/api/property/MonoMac.AVFoundation.AVMediaCharacteristic.Audible/

**Pull request closed without merging**